src/util.test.js

```diff
@@ -0,0 +1,280 @@
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import '../test/jest/__mock__';
+
+import * as utils from './util';
+
+describe('utils', () => {
+  describe('validateOptionalField', () => {
+    it('should return empty error list when there is no listName values', () => {
+      const optionalField = {
+        textFields: ['textField1'],
+        selectFields: ['selectField1'],
+        listName: 'listName',
+      };
+
+      const values = { listName: [] };
+
+      expect(utils.validateOptionalField(optionalField, values)).toEqual([]);
+    });
+
+    it('should return empty error list when validation passed for textFields and selectFields', () => {
+      const optionalField = {
+        textFields: ['textField1'],
+        selectFields: ['selectField1'],
+        listName: 'listName',
+      };
+
+      const values = {
+        listName: [{
+          textField1: 'value',
+          selectField1: 'value',
+        }],
+      };
+
+      expect(utils.validateOptionalField(optionalField, values)).toMatchObject([]);
```

Comment:
I'm guessing this statement will be true even if the array is not empty.

Reply:
there should be `toEqual`, I'll update, thanks!